File: src/solar/util/game/PlayerUtil.ts
/** @noSelf **/

/**
 * 玩家工具
 */
export default class PlayerUtil {


    /**
     * 显示文本
     * @param player
     * @param text
     * @param duration
     * @param x
     * @param y
     */
    static text(player: player, text: string, duration = 5, x: number = 0, y: number = 0) {
        DisplayTimedTextToPlayer(player, x, y, duration,
            text)
    }


    /**
     * 第一个在线用户玩家
     * 通常用作主机玩家选择难度
     */
    static firstOfUsers(): player {
        for (let i = 0; i < bj_MAX_PLAYER_SLOTS; i++) {
            let tempPlayer = Player(i);
            if (GetPlayerController(tempPlayer) == MAP_CONTROL_USER && GetPlayerSlotState(tempPlayer) == PLAYER_SLOT_STATE_PLAYING) {
                return tempPlayer;
            }
        }
    }


    /**
     * 添加金币
     * @param player
     * @param value
     */
    static addGoldState(player: player, value: number) {
        PlayerUtil.addState(player, PLAYER_STATE_RESOURCE_GOLD, value)
    }

    /**
     * 添加木材
     * @param player
     * @param value
     */
    static addLumberState(player: player, value: number) {
        PlayerUtil.addState(player, PLAYER_STATE_RESOURCE_LUMBER, value)
    }

    /**
     * 添加状态
     * @param player
     * @param whichPlayerState
     * @param value
     */
    static addState(player: player, whichPlayerState: playerstate, value: number) {
        SetPlayerState(player, whichPlayerState, GetPlayerState(player, whichPlayerState) + value);
    }

    /**
     * 获取金币
     * @param player
     */
    static getGold(player: player): number {
        return GetPlayerState(player, PLAYER_STATE_RESOURCE_GOLD)
    }

    /**
     * 获取木材
     * @param player
     */
    static getLumber(player: player): number {
        return GetPlayerState(player, PLAYER_STATE_RESOURCE_LUMBER)
    }

    /**
     * 获取人口
     * @param player
     */
    static getFoodCap(player: player): number {
        return GetPlayerState(player, PLAYER_STATE_RESOURCE_FOOD_CAP)
    }


    /**
     * 是否是在线用户玩家
     */
    static isOnlineUser(pid: number): boolean {
        return GetPlayerController(Player(pid)) == MAP_CONTROL_USER && GetPlayerSlotState(Player(pid)) == PLAYER_SLOT_STATE_PLAYING
    }
}

File: scripts/solar/util/game/PlayerUtil.lua
local ____lualib = require("lualib_bundle")
local __TS__Class = ____lualib.__TS__Class
local __TS__SourceMapTraceBack = ____lualib.__TS__SourceMapTraceBack
__TS__SourceMapTraceBack(debug.getinfo(1).short_src, {["6"] = 6,["7"] = 6,["8"] = 6,["10"] = 6,["11"] = 17,["12"] = 17,["13"] = 17,["15"] = 17,["16"] = 17,["18"] = 17,["19"] = 17,["21"] = 18,["22"] = 18,["23"] = 18,["24"] = 18,["25"] = 18,["26"] = 18,["27"] = 18,["28"] = 17,["29"] = 27,["31"] = 28,["32"] = 28,["33"] = 29,["34"] = 30,["35"] = 31,["37"] = 28,["40"] = 27,["41"] = 42,["42"] = 43,["43"] = 42,["44"] = 51,["45"] = 52,["46"] = 51,["47"] = 61,["48"] = 62,["49"] = 62,["50"] = 62,["51"] = 62,["52"] = 62,["53"] = 61,["54"] = 69,["55"] = 70,["56"] = 69,["57"] = 77,["58"] = 78,["59"] = 77,["60"] = 85,["61"] = 86,["62"] = 85,["63"] = 93,["64"] = 94,["65"] = 93,["66"] = 6});
local ____exports = {}
____exports.default = __TS__Class()
local PlayerUtil = ____exports.default
PlayerUtil.name = "PlayerUtil"
function PlayerUtil.prototype.____constructor(self)
end
function PlayerUtil.text(self, player, text, duration, x, y)
    if duration == nil then
        duration = 5
    end
    if x == nil then
        x = 0
    end
    if y == nil then
        y = 0
    end
    DisplayTimedTextToPlayer(
        player,
        x,
        y,
        duration,
        text
    )
end
function PlayerUtil.firstOfUsers(self)
    do
        local i = 0
        while i < bj_MAX_PLAYER_SLOTS do
            local tempPlayer = Player(i)
            if GetPlayerController(tempPlayer) == MAP_CONTROL_USER and GetPlayerSlotState(tempPlayer) == PLAYER_SLOT_STATE_PLAYING then
                return tempPlayer
            end
            i = i + 1
        end
    end
end
function PlayerUtil.addGoldState(self, player, value)
    ____exports.default:addState(player, PLAYER_STATE_RESOURCE_GOLD, value)
end
function PlayerUtil.addLumberState(self, player, value)
    ____exports.default:addState(player, PLAYER_STATE_RESOURCE_LUMBER, value)
end
function PlayerUtil.addState(self, player, whichPlayerState, value)
    SetPlayerState(
        player,
        whichPlayerState,
        GetPlayerState(player, whichPlayerState) + value
    )
end
function PlayerUtil.getGold(self, player)
    return GetPlayerState(player, PLAYER_STATE_RESOURCE_GOLD)
end
function PlayerUtil.getLumber(self, player)
    return GetPlayerState(player, PLAYER_STATE_RESOURCE_LUMBER)
end
function PlayerUtil.getFoodCap(self, player)
    return GetPlayerState(player, PLAYER_STATE_RESOURCE_FOOD_CAP)
end
function PlayerUtil.isOnlineUser(self, pid)
    return GetPlayerController(Player(pid)) == MAP_CONTROL_USER and GetPlayerSlotState(Player(pid)) == PLAYER_SLOT_STATE_PLAYING
end
____exports.default = PlayerUtil
return ____exports
